downloads/swan-lake-release-notes/swan-lake-preview5/RELEASE_NOTE.md

```diff
@@ -0,0 +1,270 @@
+
+An `isolated` function can also access a `final` variable if the type of the variable is a subtype of `isolated object {}`.
+
+```ballerina
+final Hotels hotels = new;
+
+isolated function getCoordinates() returns Coordinates[] => hotels.getCoordinates().toArray();
+```
+
+##### Lang lib functions to replace XML functional constructors
+
+XML functional constructor syntax has been removed. In place of XML functional constructors, `'xml:createXXX` functions should be used. 
+
+**Disallowed syntax**
+```ballerina
+'xml:Element element = 'xml:Element("elem");
+'xml:Comment comment = 'xml:Comment("comment content");
+'xml:ProcessingInstruction proceInstruc = 'xml:ProcessingInstruction("DONOT", "print this");
+'xml:Text text = 'xml:Text("this is a character sequence");
+```
+
+**Valid syntax**
+```ballerina
+'xml:Element element = 'xml:createElement("elem");
+'xml:Comment comment = 'xml:createComment("comment content");
+'xml:ProcessingInstruction proceInstruc = 'xml:createProcessingInstruction("DONOT", "print this");
+'xml:Text text = 'xml:createText("this is a character sequence");
+```
+
+##### Unified object method scope and object field scope 
+
+Object field scope and object method scope have been unified into a single scope. This means that it is no longer possible to have fields and methods with the same name.
+
+With this change, object methods can be accessed similar to object fields and the reference to `self` is captured in the resulting function pointer.
+
+```ballerina
+class Accumulator {
+    int i;
+
+    function init(int i) {
+        self.i = i;
+    }
+
+    function accumulate(int acc) {
+        self.i = self.i + acc;
+    }
+
+    function get() returns int {
+        return self.i;
+    }
+}
+
+public function main() {
+    Accumulator accumulator = new(0);
+    var acc = accumulator.accumulate; // access a method as a field
+    var getVal = accumulator.get;
+    acc(1); // similar to calling accumulator.accumulate(1);
+    acc(2);
+    int val = getVal();
+}
+```
+
+###### Improvement in accessing simple values in a JSON
+
+A non-structural value (i.e., simple value) in a JSON object can now be directly assigned to a variable of the same type without casting.
+
+For this, the field access expression should be used with the `check` expression. The simple value should belong to one of the types: `()`, `int`, `float`, `decimal`, `string`, or  `boolean`.
+
+```ballerina
+public function main() returns error? {
+    json j1 = {name: "Anne", age: 20, marks: {math: 90, language: 95}};
+    string name = check j1.name;
+    int age = check j1.age;
+}
+```
+
+##### Improved lang library functions
+
+Several changes and improvements have been introduced to the `ballerina/lang.value` lang library module.
+
+###### New `ensureType` function
+
+This method can be used to check if a value belongs to the given type similar to the type cast expression. However, instead of panicking, this method returns an error if the type test fails.
```

Comment:
```suggestion
This method can be used to check if a value belongs to the given type similar to the typecast expression. However, instead of panicking, this method returns an error if the type test fails.
```

Reply:
@MaryamZi Should we change the above?

Reply:
The spec seems to have this as two words, https://ballerina.io/spec/lang/master/#section_6.20, so I guess it is OK? Or we can just say "cast expression".

Reply:
Ack. Lets keep it as it is then.